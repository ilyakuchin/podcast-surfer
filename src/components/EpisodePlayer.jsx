import React, { Component } from "react";

export default class EpisodePlayer extends Component {
  render() {
    if (this.props.description) {
      return <div>{this.props.description}</div>;
    } else {
      return <div>Description is empty</div>;
    }
  }
}
