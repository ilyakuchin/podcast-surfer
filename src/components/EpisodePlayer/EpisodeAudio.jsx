import React, { Component } from "react";

export default class EpisodeAudio extends Component {
  render() {
    if (this.props.src) {
      return (
        <audio controls>
          <source src={this.props.src} type="audio/mp4"></source>
        </audio>
      );
    } else {
      return <div>There is no audio</div>;
    }
  }
}
