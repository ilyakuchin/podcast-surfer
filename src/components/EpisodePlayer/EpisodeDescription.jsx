import React, { Component } from "react";

export default class EpisodeDescription extends Component {
  render() {
    if (this.props.description) {
      return <div>{this.props.description}</div>;
    } else {
      return <div>Description is empty</div>;
    }
  }
}
