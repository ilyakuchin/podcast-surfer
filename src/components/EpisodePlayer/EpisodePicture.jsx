import React, { Component } from "react";

export default class EpisodePicture extends Component {
  render() {
    if (this.props.link) {
      return <img src={this.props.link}></img>;
    } else {
      return <img src="https://placekitten.com/640/360"></img>;
    }
  }
}
