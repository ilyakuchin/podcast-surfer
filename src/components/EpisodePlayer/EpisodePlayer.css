.grid {
  margin-top: 30px;
  display: grid;
  justify-items: center;
  place-items: center;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: auto;
  grid-template-areas:
    "pic header header"
    "pic audio audio"
    "description description description";
}

.pic {
  grid-area: pic;
}

.header {
  grid-area: header;
}

.description {
  grid-area: description;
  max-width: 700px;
}

.audio {
  grid-area: audio;
  width: 100%;
}
