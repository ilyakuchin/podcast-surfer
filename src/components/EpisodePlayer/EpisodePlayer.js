import React from "react";
import styled from "styled-components";

const Grid = styled.div`
  position: relative;
  bottom: 15px;
  margin-top: 30px;
  display: grid;
  grid-gap: 15px;
  grid-template-columns: 1fr 4fr;
  grid-template-rows: auto;
  grid-template-areas:
    "pic header"
    "pic audio "
    "description description";
  justify-items: center;

  @media (max-width: 600px) {
    grid-template-columns: auto;
    grid-template-areas:
      "pic"
      "header"
      "audio"
      "description";
  }
`;

const Pic = styled.img`
  grid-area: pic;
  width: 200px;
  height: 200px;
  margin-left: 10px;
  @media (max-width: 600px) {
    margin: 0;
    max-width: 50%;
    max-height: 50%;
  }
  object-fit: scale-down;
`;

const Header = styled.h2`
  grid-area: header;
  margin: 0;
  margin-left: 10px;
  margin-right: 10px;
  @media (max-width: 600px) {
    text-align: center;
  }
`;

const Audio = styled.audio`
  grid-area: audio;
  width: 95%;
`;

const Description = styled.div`
  grid-area: description;
  max-width: 700px;
  margin: 0;
  margin-left: 10px;
  margin-right: 10px;
  white-space: pre-line;
`;

export default function EpisodePlayer(props) {
  return (
    <Grid>
      <Pic src={props.location.state.image} alt="" />
      <Header>{props.location.state.name}</Header>
      <Audio controls>
        <source src={props.location.state.audio.url} type="audio/mp4"></source>
      </Audio>

      <Description>{props.location.state.description}</Description>
    </Grid>
  );
}
