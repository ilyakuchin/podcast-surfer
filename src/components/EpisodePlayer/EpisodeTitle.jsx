import React, { Component } from "react";

export default class EpisodeTitle extends Component {
  render() {
    if (this.props.title) {
      return <div>{this.props.title}</div>;
    } else {
      return <div>Episode title is missing</div>;
    }
  }
}
