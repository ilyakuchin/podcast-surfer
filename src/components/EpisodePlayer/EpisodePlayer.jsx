import React, { Component } from "react";
import EpisodeDescription from "./EpisodeDescription";

export default class EpisodePlayer extends Component {
  render() {
    return <EpisodeDescription></EpisodeDescription>;
  }
}
