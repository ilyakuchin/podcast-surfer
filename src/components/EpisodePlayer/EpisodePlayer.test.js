import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import EpisodePlayer from "./EpisodePlayer";
import EpisodeDescription from "./EpisodeDescription/EpisodeDescription";
import EpisodeTitle from "./EpisodeTitle/EpisodeTitle";
import EpisodePicture from "./EpisodePicture/EpisodePicture";
import EpisodeAudio from "./EpisodeAudio/EpisodeAudio";

configure({ adapter: new Adapter() });

describe("Episode player", () => {
  test("should return picture, title, description and audio (order sensitive)", () => {
    expect(
      shallow(
        <EpisodePlayer
          episodeInfo={{
            pictureSrc: "test",
            title: "test",
            description: "test",
            audioSrc: "test"
          }}
        ></EpisodePlayer>
      ).contains(
        <div>
          <EpisodePicture src="test"></EpisodePicture>
          <EpisodeTitle title="test"></EpisodeTitle>
          <EpisodeDescription description="test"></EpisodeDescription>
          <EpisodeAudio src="test"></EpisodeAudio>
        </div>
      )
    ).toBe(true);
  });
});
