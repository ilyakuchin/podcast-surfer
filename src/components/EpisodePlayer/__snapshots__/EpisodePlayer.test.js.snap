// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EpisodePlayer component should render correctly 1`] = `
<Redirect
  to="/"
/>
`;
